_posts/2021-04-27-rust-dev-workflow.md

```diff
@@ -0,0 +1,79 @@
+The Rust implementation of Arrow actually consists of several distinct projects, or in Rust parlance, ["crates"][3]. In addition to the Arrow crate that contains core functionality such as the in-memory format and inter-process communication, we also maintain:
+- [DataFusion][4]: an extensible in-memory query execution engine using Arrow as its format
+- [Ballista][5]: a distributed compute platform, powered by Apache Arrow and DataFusion
+
+Whilst these projects are all closely related, with many shared contributors, they're very much at different stages in their respective lifecycles. The core Arrow crate, as an implementation of a spec, has strict compatibility requirements with other versions of Arrow, and this is tested via rigorous cross-language integration tests.
+
+However, at the other end of the spectrum, DataFusion and Ballista are still nascent projects in their own right that undergo frequent backwards-incompatible changes. In the old workflow, DataFusion was released in lockstep with Arrow; because DataFusion users often need newly-contributed features or bugfixes on a tighter schedule than Arrow releases, we observed that many people in the community simply resorted to referencing our GitHub repository directly, rather than properly versioned builds on [crates.io][6], Rust's package registry.
+
+Ultimately, the decision was made to split the Rust crates into two separate repositories: [arrow-rs][7] for the core Arrow functionality, and [arrow-datafusion][8] for DataFusion and Ballista. There's still work to be done on determining the exact release workflows for the latter, but this leaves us in a much better position to meet the broader Rust community's expectations of crate versioning and stability.
+
+## Community Participation
+All Apache projects are built on volunteer contribution; it's a core principle of both the ASF and open-source software development more broadly. One point of friction that was observed in the previous workflow for the Rust community in particular was the requirement for issues to be logged in Arrow's Jira project. This step required would-be contributors to first register an account, and then receive a permissions grant to manage tickets.
+
+To streamline this process for new community members, we've taken the decision to migrate to GitHub Issues for tracking both new development work and known bugs that need addressing, and bootstrapped our new repositories by importing their respective tickets from Jira. We hope that this strikes a better balance between organisation and accessibility for prospective contributors.
+
+## Get Involved
+To further improve the onboarding flow for new Arrow contributors, we've begun the process of labelling select issues as "good first issue" in [arrow-rs][11] and [arrow-datafusion][12]. These issues are typically small in scope, but still serve as valuable contributions to the project, and help acclimatise new community members to our development workflows and tools.
```

Comment:
```suggestion
To further improve the onboarding flow for new Arrow contributors, we have started the process of labeling select issues as "good first issue" in [arrow-rs][11] and [arrow-datafusion][12]. These issues are small in scope but still serve as valuable contributions to the project. They also help new community members to get familiar with our development workflows and tools.
```
Some stylistic suggestions

Reply:
Ah, the old single vs double L, a classic in English dialect wars! Lemme scan through the blog post history, I'll standardise (heh, I'm picking a side already with that S) on either British or American English depending on which is more common. There's probably other bits that need unifying if so.

I'll try and incorporate these suggestions in some fashion, clarity is always a good goal 👍